### PR TITLE
Add support of custom agent (for example "tunnel")

### PR DESCRIPTION
**Pull request by dovyden**

### CODE REVIEW

**Comment by kaero** [comment]
Add test, pls

**Comment by dovyden** [comment]
@kaero, please, check my tests

File: test/advanced_agent.js [hunk 1]
```diff
@@ -200,5 +200,24 @@ module.exports = {
...
+        var customAgent = {
+            addRequest : function () {},
+            createSocket : function () {}
+        };
+
+        assert.strictEqual(Asker.getAgent(new Asker({ agent : customAgent })), customAgent,
+            'custom agent is returned for request with custom agent');
+    },
+
+    'Request.getAgent() should return http.globalAgent if `options.agent` is\'t Agent-like object' : function() {
```
Comment (kaero) [inline test/advanced_agent.js]: typo: `is\'t -> is not`

**Comment by kaero** [comment]
lgtm, fix typo, pls, and i'll merge it and release the module

File: test/advanced_agent.js (hunk 1 above)
Comment (dovyden) [inline test/advanced_agent.js]: fixed

**Comment by dovyden** [comment]
nice =)

**Comment by kaero** [comment]
Fixup all commits into one, please )

**Comment by dovyden** [comment]
@kaero, done =)

**Comment by kaero** [comment]
`asker 0.6.2` & `vow-asker 0.8.2`

**Pull request merged**